avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/Array_of_UNION_GenericDeserializer_585074122056792963_585074122056792963.java

```diff
@@ -74,11 +74,7 @@ public IndexedRecord deserializerecord0(Object reuse, Decoder decoder)
             decoder.readNull();
         }
         if (unionIndex1 == 1) {
-            if (record.get(0) instanceof Utf8) {
-                record.put(0, (decoder).readString(((Utf8) record.get(0))));
-            } else {
-                record.put(0, (decoder).readString(null));
-            }
+            record.put(0, (decoder).readString(((Utf8) record.get(0))));
         }
```

Comment:
Hmm... this cast doesn't seem safe. Are you sure that we can remove the `instanceof` check?

Reply:
Please check my reply above to see whether you  still have this concern or not.